kirin/core/abstract_builder.py

```diff
@@ -146,3 +83,15 @@ def build_trip_updates(self, rt_update):
         :return log_dict: dict of (k,v) to be displayed in logs and newrelic
         """
         raise NotImplementedError("Please implement this method")
+
+    def merge_trip_updates(self, navitia_vj, db_trip_update, new_trip_update):
+        # type: (Dict[unicode, Any], TripUpdate, TripUpdate) -> TripUpdate
```

Comment:
Should this be the following? I never know if this has to be specified or not.
```suggestion
        # type: (Union[None, Dict[unicode, Any]], Union[None, TripUpdate], TripUpdate) -> TripUpdate
```

Reply:
Not sure either, but we sometimes return or use None and no complaints from mypy (although those code parts are maybe not checked by it).
I didn't want to dig more (and adding types everywhere is a bigger burden than py3 migration), and I like that it expresses as clearly as possible what objects are... So I'd let it like that for now ?

Reply:
Fine with me. I'm resolving this conversation.